test/test_rb.py

```diff
@@ -0,0 +1,237 @@
+            )
+            self.assertEqual(
+                qc.metadata["qubits"],
+                tuple(exp_attributes["qubits"]),
+                "The qubits indices in the experiment metadata doesn't match to the one provided.",
+            )
+
+    def validate_circuit_data(
+        self, experiment: qe.experiment_data.ExperimentData, exp_attributes: dict
+    ):
+        """
+        Validate that the metadata of the experiment after it had run matches the one provided.
+        Args:
+
+            experiment(qiskit_experiments.experiment_data.ExperimentData): The experiment
+            data and results after it run.
+            exp_attributes (dict): A dictionary with the experiment variable ands values
+        """
+        for ind, data in enumerate(experiment.data):
+            experiment_information = data["metadata"]
+            self.assertEqual(
+                experiment_information["xdata"],
+                exp_attributes["lengths"][ind],
+                "The number of gates in the experiment doesn't match to the one in the metadata.",
+            )
+            self.assertEqual(
+                experiment_information["qubits"],
+                exp_attributes["qubits"],
+                "The qubits indices in the experiment doesn't match to the one in the metadata.",
+            )
+
+    def _exp_data_prpeties(self):
+        """
+        Return a list of dictionaries that contains invalid experiment propeties to check errors.
+        """
+        exp_data_list = [
+            {"qubits": [3, 3], "lengths": [1, 3, 5, 7, 9], "num_samples": 0, "seed": 100},
+            {"qubits": [-1], "lengths": [1, 3, 5, 7, 9], "num_samples": 1, "seed": 100},
+            {"qubits": [0, 1], "lengths": [1, 3, 5, -7, 9], "num_samples": 1, "seed": 100},
+            {"qubits": [0, 1], "lengths": [1, 3, 5, 7, 9], "num_samples": -4, "seed": 100},
+            {"qubits": [0, 1], "lengths": [1, 3, 5, 7, 9], "num_samples": 0, "seed": 100},
+            {"qubits": [0, 1], "lengths": [1, 5, 5, 5, 9], "num_samples": 0, "seed": 100},
+        ]
+        return exp_data_list
+
+    def _test_input(self):
+        """
+        Check that errors emerge when invalid input is given to the RB experiment.
+        """
+        exp_data_list = self._exp_data_prpeties()
+        rb = qe.randomized_benchmarking
+        for exp_data in exp_data_list:
+            self.assertRaises(
+                QiskitError,
+                rb.RBExperiment,
+                exp_data["qubits"],
+                exp_data["lengths"],
+                num_samples=exp_data["num_samples"],
+                seed=exp_data["seed"],
+            )
+
+    def test_RB_circuits(self):
+        """
+        Run the RB test for the circuits (checking the metadata, parameters and functionallity
+        of the experiment.
+        """
+        self._test_input()
+        self.rb_parameters_one_qubit()
```

Comment:
the only difference between these 3 functions is the qubits parameter.
consider writing it by one function and use ddt and combine for the parameters 
see for example: 
https://github.com/Qiskit/qiskit-terra/blob/master/test/python/quantum_info/operators/symplectic/test_clifford.py

Reply:
I agree with Shelly. It is better to keep separate tests with separate names, but have all the tests call the same main testing function with different sets of parameters (I think the lengths and seed should be parameters as well, not just qubits).

Reply:
Changed it. now the code have the following decorator:
@data(qubits=([3], [4, 7], [0, 5, 3]))